### ISSUE TITLE
Comments included as part of logfile path

### ISSUE THREAD

**Issue by mnaberez** [Supervisor/supervisor#1229]
@sdanbury wrote in https://github.com/Supervisor/supervisor/issues/120#issuecomment-483318092:

> Haven't seen this for a couple of years, but it has randomly started appearing again.
>
> $ sudo supervisord -c /etc/supervisor/supervisord.conf
Error: The directory named as part of the path /var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log) does not exist
> For help, use /usr/local/bin/supervisord -h
>
> I have made sure the /var/log/supervisor/supervisord.log file exists and have even given permissions of 777 to test, and it is still happening. Any ideas?

**Comment by mnaberez** [comment]
> Error: The directory named as part of the path /var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log) does not exist

@sdanbury It seems to think the comments are part of the log file path.

- What version of Supervisor are you using (`supervisord --version`)?
- What version of Python are you using (`python -V`)?
- What are the contents of the config file (`cat /etc/supervisor/supervisord.conf`)?

**Comment by sdanbury** [comment]
Thanks for creating a separate issue, apologies.

```
$ supervisord --version
4.0.1
```
```
$ python -V
Python 2.7.6
```
```
$ cat /etc/supervisor/supervisord.conf
; supervisor config file

[unix_http_server]
file=/var/run/supervisor.sock   ; (the path to the socket file)
chmod=0700                       ; sockef file mode (default 0700)

[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)
pidfile=/var/run/supervisor/supervisord.pid ; (supervisord pidfile;default supervisord.pid)
childlogdir=/var/log            ; ('AUTO' child log dir, default $TEMP)

; the below section must remain in the config file for RPC
; (supervisorctl/web interface) to work, additional interfaces may be
; added by defining them in separate rpcinterface: sections
[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[supervisorctl]
serverurl=unix:///var/run/supervisor.sock ; use a unix:// URL  for a unix socket

; The [include] section can just contain the "files" setting.  This
; setting can list multiple files (separated by whitespace or
; newlines).  It can also contain wildcards.  The filenames are
; interpreted as relative to this file.  Included files *cannot*
; include files themselves.
```

**Comment by sdanbury** [comment]
Removed the comments for now, and it worked as expected. Maybe some sort of white space issue? Not too sure.

**Comment by mnaberez** [comment]
I copy/pasted the `logfile=` line from the config [above](https://github.com/Supervisor/supervisor/issues/1229#issuecomment-483542885)  and wasn't able to see the error:

```
$ cat supervisord.conf
[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)

$ supervisord -n -c supervisord.conf
2019-04-16 10:18:57,154 INFO Increased RLIMIT_NOFILE limit to 1024
2019-04-16 10:18:57,155 INFO supervisord started with pid 6671
```

If I remove the whitespace before the semicolon, I do see the error:

```
$ cat supervisord.conf 
[supervisord]
logfile=/var/log/supervisor/supervisord.log; (main log file;default $CWD/supervisord.log)

$ supervisord -n -c supervisord.conf 
Error: The directory named as part of the path /var/log/supervisor/supervisord.log; (main log file;default $CWD/supervisord.log) does not exist
For help, use /usr/local/bin/supervisord -h
```

That's the [expected behavior](https://github.com/Supervisor/supervisor/blob/4.0.1/supervisor/skel/sample.conf#L11) due to how Python's `ConfigParser` works.

However, your config file above has whitespace before the semicolon.  Can you confirm that you did have whitespace before the semicolon when you saw the error?

**Comment by BillMedernach** [comment]
I'm running into the same error on the `minfds` config param where the comment is getting parsed as part of the configuration value. If I downgrade from 4.0.0 or 4.0.1 to 3.3.5 the issue goes away.

**Comment by sdanbury** [comment]
@mnaberez thank you for looking into it, much appreciated. I can confirm that it had whitespace before the semicolon when I saw the error.

**Comment by mnaberez** [comment]
@BillMedernach It might be related but I don't think it can be the same error, since the message `Error: The directory named as part of the path` should only occur with logfile paths.  I created #1230 for your issue, please follow up with me there.

**Comment by mnaberez** [comment]
> I can confirm that it had whitespace before the semicolon when I saw the error.

@sdanbury Could you please try to reproduce it and let me know if you're able to make it happen again?

```
$ cat supervisord.conf
[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)

$ supervisord -n -c supervisord.conf
2019-04-16 10:18:57,154 INFO Increased RLIMIT_NOFILE limit to 1024
2019-04-16 10:18:57,155 INFO supervisord started with pid 6671
```

I've tried the above with spaces and tabs before the `;` and wasn't able to see the error.  I can only get the error to occur when I remove the whitespace from before the `;`.  I am hoping that I can find a way to reproduce it locally so I can troubleshoot it.

**Comment by sdanbury** [comment]
@mnaberez I just tried to recreate it outside of the VM I see it happening on and I can't seem to recreate.

```
$ docker run -it ubuntu:14.04 bash
# apt-get update

# apt-get install -y python-minimal python-pip

# python -V
Python 2.7.6

# echo "[supervisord]" >> supervisord.conf
# echo "logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)" >> supervisord.conf

# supervisord -n -c supervisord.conf
```

No issue. Is there anything else I can check for on the VM it is happening on? To be honest, Ubuntu 14.04 and Python 2.7.6 is a really old stack now, so in my case, I would be happy to close. But willing to dig deeper if there is anything else you want to know?

**Comment by mnaberez** [comment]
I [added tests](https://github.com/Supervisor/supervisor/blob/d3f48a8de28f1e70a9fabaef80b3dd79d05cec70/supervisor/tests/test_options.py#L706-L729) for this case (`logfile`) and also `minfds` from #1230 and it [passes on CI](https://travis-ci.org/Supervisor/supervisor/builds/521313666).  The CI environment is [using 2.7.14](https://travis-ci.org/Supervisor/supervisor/jobs/521313670#L433).  I was not able to reproduce this locally yet.  

> No issue. Is there anything else I can check for on the VM it is happening on?

If you can still get it to happen on that VM, could you please attach the actual `supervisord.conf` file (drag the file into GitHub comment box), or alternatively paste the output of either `hexdump supervisord.conf` or `base64 supervisord.conf`?  I will try to reproduce it using your byte-for-byte exact config file.

**Comment by mnaberez** [comment]
I believe this is the same issue #1230 and that it has been fixed in 2c2cd3567dc22c53d7d73442c83a9390cd210259.  I am going to close this issue for now.  Thank you for your help!
